src/shared/components/Banners/Banner/Banner.tsx

```diff
@@ -0,0 +1,59 @@
+import { Alert } from '@mui/material';
+import { useEffect, useState } from 'react';
+import useWindowFocus from 'use-window-focus';
+
+import { Svg } from 'shared/components/Svg';
+import { StyledClearedButton } from 'shared/styles';
+
+import { BannerProps } from './Banner.types';
+
+export const Banner = ({
+  children,
+  duration = 5000,
+  onClose,
+  hasCloseButton = !!onClose,
+  severity,
+}: BannerProps) => {
+  let timeoutId: NodeJS.Timeout | undefined;
+  const [isHovering, setIsHovering] = useState(false);
+  const isWindowFocused = useWindowFocus();
+
+  // Close banner on timeout only while window is focused & banner not hovered
+  // (a11y behavior adapted from MUI SnackBar)
+  useEffect(() => {
+    if (!duration || !onClose || isHovering || !isWindowFocused) return;
+
+    timeoutId = setTimeout(onClose, duration);
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [duration, isHovering, isWindowFocused, onClose]);
+
+  return (
+    <Alert
+      iconMapping={{
+        info: getSvg('more-info-filled'),
+        success: getSvg('check-circle'),
+        warning: getSvg('exclamation-circle'),
+        error: getSvg('exclamation-octagon'),
+      }}
+      components={{
+        CloseButton: StyledClearedButton,
+      }}
+      componentsProps={{
```

Comment:
Since `components` and `componentsProps `will be deprecated in the future please use `slots` and `slotProps ` accordingly.

https://mui.com/material-ui/api/alert/#alert-prop-componentsProps
https://mui.com/material-ui/api/alert/#alert-prop-slotProps

Reply:
Good catch, thanks!

Reply:
Thank you! could you please update `components` to `slots` as well?

Reply:
Of course, my bad… 🤦🏻 Done!